src/contact_info.rs

```diff
@@ -159,6 +164,24 @@ impl ContactInfo {
     pub fn is_valid_address(addr: &SocketAddr) -> bool {
         (addr.port() != 0) && Self::is_valid_ip(addr.ip())
     }
+    pub fn get_sign_data(&self) -> Vec<u8> {
```

Comment:
can you break this out into a singe signed serialized structure?  @garious i am not a huge fan of this pattern from Transaction, because of how error prone it is.

Reply:
Yes, Transaction should have a TransactionData struct that includes everything but its signatures.

Reply:
Alright I can. I'll end up sticking signature into the Protocol enums instead of into the individual crdsvalues. That way I can just use default serialization for the values. Sounds good? Or should I do what @garious suggested with a data inner struct?

Reply:
Oh that's right! Okay then I need to use the inner data structs to avoid manually serializing. Thanks for pointing that out. Got carried away trying to simplify...